### DISCUSSION TITLE
open devtools with `parcel serve`

### DISCUSSION THREAD

**Discussion by avalanche1** [parcel-bundler/parcel#7499] [Q&A] [answered]
Is it possible to automatically open devtools, when running `parcel serve` and then opeining localhost:1234 in browser?

**Comment by mischnic** [comment, marked as answer]
There is apparently no JS api to do this. The only way I can think of is (on macOS) using AppleScript to trigger the "View > Developer > Devtools" menu (not sure what the situation is on other platforms, but the approach would have to similar). This is also how create-react-app is able to reuse an existing browser tab: https://github.com/facebook/create-react-app/blob/bb64e31a81eb12d688c14713dce812143688750a/packages/react-dev-utils/openBrowser.js#L68-L74
